### PR TITLE
[MAISTRA-462] Put the 'member-of' label in the control plane namespace

### PR DESCRIPTION
**Pull request by jwendell**
So that Kiali knows about it.

### CODE REVIEW

File: pkg/controller/servicemesh/controlplane/reconciler.go [hunk 1]
```diff
@@ -65,11 +66,16 @@ func (r *ControlPlaneReconciler) Reconcile() (reconcile.Result, error) {
 		if namespace.Labels == nil {
 			namespace.Labels = map[string]string{}
 		}
+
+		// Label the control plane namespace so that Kiali knows about it
+		namespace.Labels[common.MemberOfKey] = r.Instance.Namespace
+
```
Comment (rcernich) [inline pkg/controller/servicemesh/controlplane/reconciler.go:72]: We should probably remove these labels during uninstall/delete.

Comment (jwendell) [reply pkg/controller/servicemesh/controlplane/reconciler.go:72]: Done

File: pkg/controller/servicemesh/controlplane/reconciler.go [hunk 2]
```diff
@@ -65,11 +67,17 @@ func (r *ControlPlaneReconciler) Reconcile() (reconcile.Result, error) {
...
-			r.Log.Info("Adding maistra.io/ignore-namespace=ignore label to Request.Namespace")
-			namespace.Labels["maistra.io/ignore-namespace"] = "ignore"
-			err = r.Client.Update(context.TODO(), namespace)
+
+		// Label the control plane namespace so that Kiali knows about it
+		r.Log.Info(fmt.Sprintf("Adding %s label to namespace %s", common.MemberOfKey, r.Instance.Namespace))
+		namespace.Labels[common.MemberOfKey] = r.Instance.Namespace
+
+		if label, ok := namespace.Labels[common.IgnoreNamespace]; !ok || label != "ignore" {
+			r.Log.Info(fmt.Sprintf("Adding %s=ignore label to namespace %s", common.IgnoreNamespace, r.Instance.Namespace))
```
Comment (rcernich) [inline pkg/controller/servicemesh/controlplane/reconciler.go:81]: this test prevents updating the namespace if the label already exists.  this code will be executed for every update to the CR, so we save an api call if the labels already exist.

File: pkg/controller/servicemesh/controlplane/reconciler.go [hunk 3]
```diff
@@ -65,11 +67,17 @@ func (r *ControlPlaneReconciler) Reconcile() (reconcile.Result, error) {
...
 			namespace.Labels = map[string]string{}
 		}
-		if label, ok := namespace.Labels["maistra.io/ignore-namespace"]; !ok || label != "ignore" {
-			r.Log.Info("Adding maistra.io/ignore-namespace=ignore label to Request.Namespace")
-			namespace.Labels["maistra.io/ignore-namespace"] = "ignore"
-			err = r.Client.Update(context.TODO(), namespace)
+
+		// Label the control plane namespace so that Kiali knows about it
+		r.Log.Info(fmt.Sprintf("Adding %s label to namespace %s", common.MemberOfKey, r.Instance.Namespace))
+		namespace.Labels[common.MemberOfKey] = r.Instance.Namespace
```
Comment (rcernich) [inline pkg/controller/servicemesh/controlplane/reconciler.go]: This should be part of the test below too.  We don't want to update the namespace if we don't have to.

File: pkg/controller/servicemesh/controlplane/reconciler.go (hunk 2 above)
Comment (jwendell) [reply pkg/controller/servicemesh/controlplane/reconciler.go:81]: Thanks for the clarification

File: pkg/controller/servicemesh/controlplane/reconciler.go (hunk 3 above)
Comment (jwendell) [reply pkg/controller/servicemesh/controlplane/reconciler.go]: Done, thanks.

**Comment by jwendell** [comment]
Closing as absolete

**Pull request closed without merging**